gammapy/scripts/cta_sensitivity.py

```diff
@@ -0,0 +1,250 @@
+    - livetime: u.h
+        Livetime (object with the units of time), e.g. 5*u.h
+Optionnal inputs:
+    - slope: float (>0)
+        Index of the spectral shape (Power-law)
+    - alpha: float
+        On/OFF normalisation
+    - sigma: float
+        minimun significance
+    - gamma_min: float
+        Minimum number of gamma-rays
+    - bkg_sys: float
+        Fraction of Background systematics relative to the number of ON counts
+    - verbosity: int
+        level of print out [int: 0/1/2(debug)]. In the debug mode, stats for each bin are printed and the computed \
+        sensitivity curve is superposed with the one stored in the IRF fits file
+
+"""
+
+# Inputs :
+
+# IRFs :
+#  - taux de fond en énergie reconstruite
+#  - matrice de dispersion en énergie
+#  - surface efficace en énergie vraie
+# Paramètres :
+#  - nsigma, alpha, temps, ngamma_min et systématiques fond
+
+# Taux de fond, calcul de nombre de gamma
+
+# calcul du flux avec la double intégrale et le nombre de gamma
+
+import astropy.units as u
+import numpy as np
+import matplotlib.pyplot as plt
+import sys
+
+from gammapy.stats import significance_on_off
+from gammapy.spectrum.models import PowerLaw
+from gammapy.spectrum.utils import CountsPredictor
+from gammapy.scripts import CTAPerf
+from gammapy.utils.scripts import get_parser, set_up_logging_from_args
+
+
+class cta_sensi_estim(object):
+    """
+    Parameters:
+        - filename: str
+            IRF filename (full path, in general there are here: ../datasets/cta/)
+            e.g. ../datasets/cta/CTA-Performance-South-20170323/CTA-Performance-South-20deg-30m_20170323.fits
+        - livetime: u.h
+            Livetime (object with the units of time), e.g. 5*u.h
+    Optional Parameters:
+        - slope: float (>0)
+            Index of the spectral shape (Power-law)
+        - alpha: float
+            On/OFF normalisation
+        - sigma: float
+            minimun significance
+        - gamma_min: float
+            Minimum number of gamma-rays
+        - bkg_sys: float
+            Fraction of Background systematics relative to the number of ON counts
+        - verbosity: int
+            level of print out [int: 0/1/2(debug)]. In the debug mode, stats for each bin are printed and the computed \
+            sensitivity curve is superposed with the one stored in the IRF fits file
+    """
+
+    def __init__(self,irffile,livetime,slope=2.,alpha=0.2,sigma=5.,gamma_min=10.,bkg_sys=0.05, verbosity=0):
+        self.filename = irffile
+        self.livetime = (livetime*u.Unit('h')).to('s')
+        self.slope = slope
+        self.alpha = alpha
+        self.sigma = sigma
+        self.gamma_min = gamma_min
+        self.bkg_sys = bkg_sys
+        self.verbosity = verbosity
+        self.energy = None
+        self.diff_sens = None
+        # Reading of the IRFs
+        self.cta_perf = CTAPerf.read(self.filename)
```

Comment:
Usually the coding pattern we have is that `__init__` takes the data members it needs and doesn't do file I/O.
So here, I would suggest to take a `CTAPerf` object as input and just let the caller construct it before initialising this class.